core/blockchain/blockchain_gateway.py

```diff
@@ -1,164 +1,157 @@
-                self.start_listening(event_filter))
-            if filtered_diffs:
-                check = list(map(handler, filtered_diffs))
-            else:
-                check = None
-        finally:
-            loop.close()
-        return check
-
-    def handle_decentralized_learning_trainer(self, tx: dict) -> None:
-        """
-        This callback will be triggered by the Listener if it finds the 
-        method signature it's looking for.
-        The parameters (model weights, model config) will be downloaded 
-        and put into the optimizer initially. So the optimizer knows this info.
-        """
-        logging.info("handling decentralized learning... {}".format(tx))
-        assert TxEnum.KEY.name in tx
-        key = tx.get(TxEnum.KEY.name)
-        value = tx.get(TxEnum.CONTENT.name)
-        # args = {TxEnum.KEY.name: key, TxEnum.CONTENT.name: ipfs_to_content(self.client, value)}
-        self.communication_manager.inform(RawEventTypes.NEW_SESSION.name, value)
-        # self.listen_new_weights()
-
-    def handle_new_weights(self, tx: dict):
-        """
-        handle_new_weights() method downloads weights and does something with
-        it. This callback is triggered by the Listener when it sees new weights 
-        intended for its node ID. Downloads the weights, and gives them to the
-        comm. mgr which gives them to the relevant optimizer 
-        -which should do the moving average.
-        """
-        # TODO: Upon CommMgr PR being merged, update this with the appropriate args
-        logging.info("handling new weights...{}".format(tx))
-        key = tx.get(TxEnum.KEY.name)
-        value = tx.get(TxEnum.CONTENT.name)
-        args = {TxEnum.KEY.name: key, TxEnum.CONTENT.name: ipfs_to_content(self.client, value)}
-        # TODO: Put into in-memory datastore.
-        self.communication_manager.inform("NEW_INFO",
-                                            args)
-        # TODO: Uncomment below line as soon as we can get Travis to work with these tests
-        # self.listen_new_weights()
-
-    def handle_terminate(self) -> list:
-        self.communication_manager.inform("TERMINATE", None)
-
-    def listen_decentralized_learning(self) -> list:
-        """
-        Polls blockchain for node ID in decentralized_learning() method
-        signature decentralized_learning(...<node_ids>...) method signature
-        will be the key on the blockchain; listener should look for this, and
-        if the method signature contains its node id, it will trigger a
-        callback
-        """
-        def filter(tx: dict) -> str:
-            return tx.get(TxEnum.KEY.name) == tx.get(TxEnum.CONTENT.name)
-        return self.filter_set(filter,
-                                self.handle_decentralized_learning_trainer)
-
-    def listen_new_weights(self) -> None:
+        logging.info("Starting cron...")
+        self._listen_as_event(period_in_mins, 
```

Comment:
Nitpick but if you're going to separate arguments to the function in separate lines, it's more readable if you separate everything out. 

```
self._listen_as_event(
                period_in_mins, 
                self._handle_new_session_creation,
                self._filter_new_session
)
```

Reply:
Done